src/handle.jl

```diff
@@ -29,25 +29,33 @@ typealias KINMemPtr Ptr{KINMem}
    Manages automatic destruction of the referenced objects when it is
    no longer in use.
 """
-immutable Handle{T <: AbstractSundialsObject}
+immutable Handle{T}
     ptr_ref::Ref{Ptr{T}} # pointer to a pointer
 
-    @compat function (::Type{Handle}){T}(ptr::Ptr{T})
+    @compat function (::Type{Handle}){T<: AbstractSundialsObject}(ptr::Ptr{T})
         (ptr == C_NULL) && throw(ArgumentError("Null pointer passed to Handle()"))
         h = new{T}(Ref{Ptr{T}}(ptr))
         finalizer(h.ptr_ref, release_handle)
         return h
     end
+    @compat function (::Type{Handle})(ptr::Ptr{Void})
+        h = new{Void}(Ref{Ptr{Void}}(ptr))
+        finalizer(h.ptr_ref, release_handle)
+        return h
+    end
 end
 
 Base.convert{T}(::Type{Ptr{T}}, h::Handle{T}) = h.ptr_ref[]
 Base.convert{T}(::Type{Ptr{Ptr{T}}}, h::Handle{T}) = convert(Ptr{Ptr{T}}, h.ptr_ref[])
 
 release_handle{T}(ptr_ref::Ref{Ptr{T}}) = throw(MethodError("Freeing objects of type $T not supported"))
+release_handle(ptr_ref::Ref{Ptr{Void}}) = nothing
```

Comment:
I think we just need to introduce the `C_NULL` check in each `release_handle()`.